### PR TITLE
[MOB-20427] - don't show fullscreenmessage with empty html

### PR DESCRIPTION
**Pull request by sbenedicadb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift [hunk 1]
```diff
@@ -345,6 +345,14 @@
         }
 
         private func displayWithAnimation(webView: WKWebView) {
+            // MOB-20427 - only display the WKWebView if we have html to show
```
Comment (cdhoffmann) [inline AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift:348]: Is there a reason we can't do this check earlier at the beginning of the `handleShouldShow`? That way you don't inform the delegates if there is no HTML to begin with?

**Review by cdhoffmann — APPROVED** [review]

File: AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift (hunk 1 above)
Comment (cdhoffmann) [reply AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift:348]: > Is there a reason we can't do this check earlier at the beginning of the `handleShouldShow`? That way you don't inform the delegates if there is no HTML to begin with?
From Steve internally:
1. they want holdout tracking to behave like normal - i.e. they want "show" events sent to ajo even for a holdout experience (which is why the html is empty)
2. if the change gets put elsewhere, someone could call displayWithAnimation in the future without making the check for empty html.  this is the safer place to put the check as it covers all current and future scenarios

Comment (sbenedicadb) [reply AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift:348]: two reasons the code is where it is:

1. we want holdout tracking to behave like normal message tracking - i.e. we want "show" events sent to ajo even for a holdout experience (which is why the html is empty to begin with)
2. if the change gets put elsewhere, someone could call displayWithAnimation in the future without making the check for empty html.  this is the safer place to put the check as it covers all current and future scenarios

Comment (cdhoffmann) [reply AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift:348]: > Is there a reason we can't do this check earlier at the beginning of the `handleShouldShow`? That way you don't inform the delegates if there is no HTML to begin with?
From Steve internally:
1. they want holdout tracking to behave like normal - i.e. they want "show" events sent to ajo even for a holdout experience (which is why the html is empty)
2. if the change gets put elsewhere, someone could call displayWithAnimation in the future without making the check for empty html.  this is the safer place to put the check as it covers all current and future scenarios

**Pull request merged**